asserts/internal/grouping.go

```diff
@@ -98,32 +107,50 @@ func (gr *Groupings) AddTo(g *Grouping, group uint16) error {
 		g.elems = []uint16{group}
 		return nil
 	}
-	// TODO: support using a bit-set representation after the size point
-	// where the space cost is the same
-	found, j := g.search(group)
+	found, j := gr.search(g, group)
 	if found {
 		return nil
 	}
+	newsize := g.size + 1
+	if newsize > gr.bitsetThreshold {
+		// switching to a bit-set representation after the size point
+		// where the space cost is the same, the representation uses
+		// bitsetThreshold-many 16-bits words stored in elems
```

Comment:
Is there a specific reason to go with 16 bit words? Either 8 bit or 32bit / 64bit would probably be more "natural".